bridge-stp.in

```diff
@@ -212,7 +212,7 @@ case "$action" in
             # Add bridge to mstpd and configure.
             echo "Adding/configuring bridge '$bridge' ..."
             "$mstpctl" addbridge "$bridge" || continue
-            if [ -n "$config_cmd" ]; then
+            if [ -x "$config_cmd" ]; then
```

Comment:
I vote 3., but the problem with type is that it's not very portable and it returns human readable descriptions, as does command -V. busybox sh  (type false --> 'false is a shell builtin'. type -t returns the command/function/path. It even returns true if the tested command is missing x permission. Although I think if someone sets config_cmd to a non executable thing, such as a directory, it's ok if trying to execute it gives an error.

So, I vote for 3 and
if $(type "$config_cmd" >/dev/null) i.e leave out the [ -x "$config_cmd" ].